### ISSUE TITLE
Wrapping byte/char as bool causes invalid bool value

### ISSUE THREAD

**Issue by kwesolowski** [sharpdx/SharpDX#648]
Hello,

In class JoystickState there is array of `bool` representing Button state:
https://github.com/sharpdx/SharpDX/blob/e6bb4e3b5a8cfe27985a4a7f5346959d8076754e/Source/SharpDX.DirectInput/JoystickState.cs#L73

However those values are not always 0 or 1 - we stepped upon value 0x80 representing bool. While it is displayed as true by debugger and treated as true when used inside if statement, when it is "anded" with second value "bitwise and" is used on x86. Therefore we have such funny code:

```
bool a = true; //a is true, actually 0x01
bool b = buttons[20]; //b is true, actually 0x80
bool c = a && b; //c is false, becouse C# compiler assumes booleans are either 0 or 1, and && is implemented as "bitwise and" in x86 assembly.
```

**Comment by xoofx** [comment]
good catch!

**Comment by BrannonKing** [comment]
+1. Any progress on fixing this?

**Comment by xoofx** [comment]
Fixed in the commit 333d1c2
